content/news/2023-03-04-bevy-0.10/index.md

```diff
@@ -0,0 +1,1546 @@
+
+1. Every system can belong to at most one base set.
+2. Systems that do not specify a base set will be added to the default base set for the schedule (if the schedule has one).
+
+```rust
+// You can add new base sets to any built-in ones
+#[derive(Debug, Hash, PartialEq, Eq, Clone, SystemSet)]
+#[system_set(base)]
+enum MyBaseSet {
+    Early,
+    Late,
+}
+
+app
+    // This ends up in CoreSet::Update by default
+    .add_system(no_base_set)
+    // You must use .in_base_set rather than .in_set for explicitness
+    // This is a high-impact decision!
+    .add_system(post_update.in_base_set(CoreSet::PostUpdate))
+    // Look, it works!
+    .add_system(custom_base_set.in_base_set(MyBaseSet::Early))
+    // Ordering your base sets relative to CoreSet is probably wise
+    .configure_set(MyBaseSet::Early.before(CoreSet::Update))
+    .configure_set(MyBaseSet::Late.after(CoreSet::Update));
+```
+
+Pretty simple, but what does this buy us?
+First, it gives you a clear hook to impose, reason about and visualize high level structure to your schedule. Yearning for a linear, stage-like design? Just order your base sets!
+Secondly, it allows Bevy to set good default behavior for systems added by users, without removing their control.
+
+Let me tell you a story, set in a world where all of Mr. Straw Man's points above are true, and no default set is added.
+
+1. A new user adds the `make_player_run` system to their app.
+2. Sometimes this system runs before input handling, leading to randomly dropped inputs. Sometimes it runs after rendering, leading to strange flickers.
+3. After much frustration, the user discovers that these are due to "system execution order ambiguities".
+4. The user runs a specialized tool, digs into the source code of the engine, figures out what order their system should run in relative to the engine's system sets, and then continues on their merry way, doing this for each new system.
+5. Bevy (or one of their third-party plugins) updates, breaking all of our poor users system ordering once again.
+
+In practice, there are three broad classes of systems: gameplay logic (the majority of all end user systems), stuff that needs to happen before gameplay logic (like event cleanup and input handling) and stuff that needs to happen after gameplay logic (like rendering and audio).
+
+By broadly ordering the schedule via base sets, we hope that Bevy apps can have good default behavior and clear high level structure without compromising on the scheduling flexibility and explicitness that advanced users crave.
+Let us know how it works out for you!
+
+### Polish Matters
+
+As part of this work, we've taken the time to listen to our users and fix some small but high-impact things about how scheduling works.
+
+Compare the following options for adding and ordering four systems, one after the other.
+
+:coffee: **Enterprise-grade** :coffee::
+
+```rust
+#[derive(SystemSet, PartialEq, Eq, Clone, Copy, Hash, Debug)]
+#[allow(missing_docs)]
```

Comment:
```suggestion
```

Reply:
This was a joke ;) Fine to cut it though.

Reply:
Oops, went over my head.